### DISCUSSION TITLE
Request for seamless installation abilities through package managers

### DISCUSSION THREAD

**Discussion by aiavci** [qpdf/qpdf#524] [Q&A]
### Question
What's the reason for not providing the latest versions through `yum` or `apt-get`? `yum install qpdf` installs version 5.0.1?

**Comment by dreua** [comment]
I think this is the wrong place to ask this question. The distributions have package maintainers and rules for updates, qpdf does not have much influence on that afaik.

**Comment by dreua** [comment]
Here is an overview of qpdf versions in different distributions. Note that package maintainers may back port certain features and security fixes.

https://repology.org/project/qpdf/versions

I personally am very happy with Fedora. It gives me a good balance of up to date packages and stability.

Idea for a workaround: You could use a docker image. Maybe qpdf could provide guidance on how to do this.

**Comment by jberkenbilt** [comment]
There's also an [AppImage](https://appimage.github.io/qpdf/) for qpdf. This is a self-contained executable that should work on any reasonably recent Linux distribution. It's similar to a docker image in that it encapsulates all the required libraries, but it's a bit lighter weight than a docker image and doesn't require any additional setup to run. If you're using Ubuntu, there's a [ppa](https://launchpad.net/~qpdf/+archive/ubuntu/qpdf) that I maintain. I am the Debian maintainer for qpdf, so these are the same packages as will eventually be in the latest Ubuntu release. Whenever I release qpdf, I upload a new Debian version and then also update the Ubuntu ppa. I am not part of the RedHat/Fedora community, but in my experience, Fedora stays quite up to date, as do the other distributions. Older, long-term support, enterprise-focused distributions don't general take upgrades but rather prefer to backport security fixes. Thankfully qpdf has had very few security issues and hasn't had any for a long time.

If there is demand, I could provide a docker image for qpdf.

**Comment by aiavci** [comment]
Thank you for the responses. I am trying to install QPDF on Amazon Linux 2, which I believe may be CentOS.

This may be becoming a StackOverflow question but how would QPDF users expect to install QPDF on Elastic Beanstalk for example? 

If the latest qpdf version were available through yum, this could have been through an `.ebextensions` `<any-file-name>*.config` file with the simple content below:

```
packages: 
  yum:
    make: []
    gcc: []
    zlib: []
    libjpeg: []
    qpdf: []
```

Thanks

**Comment by dreua** [comment]
> If the latest qpdf version were available through yum

It is, at least on Fedora 32 I get 10.2.0. In other words: "available through yum" misses the point. Yum is just a software fronted, what matters is it's configuration and the system it is running on. Run `lsb_release -a` to get more information on the latter.

```
$ sudo yum install qpdf
Last metadata expiration check: 0:00:30 ago on So 25 Apr 2021 18:24:35 CEST.
Package qpdf-10.2.0-1.fc32.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

However, I don't feel like giving free support for the very commercial and capitalistic Amazon services, therefore I'm going to unsubscribe here. Nothing personal against you but something personal against Amazon. (You may email me for paid CentOS-support.)

**Comment by aiavci** [comment]
That makes sense. Thank you for the response.

`yum install qpdf` (Or it's config equivalent) getting me an outdated version must be due to an unrelated issue for me then which makes my request for a change invalid in the first place.

**Comment by jberkenbilt** [comment]
@dreua By the way, thanks for your responses to this issue. It's great when members of the community help out. I was completely unaware of repology.org, and looking at that link was quite interesting. I'm going to convert this issue to a discussion. Thanks to all who participated.
